projects/igniteui-angular/src/lib/directives/mask/mask.directive.ts

```diff
@@ -153,8 +153,9 @@ export class IgxMaskDirective implements OnInit, AfterViewChecked, ControlValueA
 
     /** @hidden */
     public ngOnInit(): void {
-        this.renderer.setAttribute(this.nativeElement, 'placeholder',
-            this.placeholder ? this.placeholder : this.maskOptions.format);
+        if (!this.nativeElement.placeholder) {
+          this.renderer.setAttribute(this.nativeElement, 'placeholder', this.maskOptions.format);
```

Comment:
But the placeholder option is just deprecated and not removed yet, correct? Cause you're stop using it completely in this change, which means that you're effectively removing it.

Reply:
The mask's `placeholder` is merely a proxy to the native attribute - added specifically to intercept it as input and then manually assign the native one anyways.

The deprecation removed the input decorator (leaving the native assign to go through). This fixes the mask self-triggering the deprecation warning.

And yes, the idea is to remove the native public prop or at least make it private.

Reply:
@kdinev let's remove the warning with that fix, we will remove all deprecated properties for 10.0.0 in another PR?